app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -47,6 +47,12 @@ def deployment_spec(opts)
       target[SPEC_KEY_MAPPING[r]] = resource.ems_ref if resource.present?
     end
 
+    if target["folder_id"].blank? && opts[:datacenter_id].present?
```

Comment:
Should this be `:folder_id` or is `target` a HashWithIndifferentAccess?  Either way I think we should dereference the hash consistently

Reply:
Ah I missed that this was checking `target` then `opts`

Reply:
Vshpere API expects a hash with keys in string. Our internal hash `opts` has keys in symbol.